README.md

```diff
@@ -811,6 +813,23 @@ RuboCop::RakeTask.new(:rubocop) do |task|
 end
 ```
 
+## Caching
+
+Large projects containing hundreds or even thousands of files can take
+a really long time to inspect, but RuboCop has functionality to
+mitigate this problem. A result caching mechanism is activated if the
+configuration parameter `AllCops: UseCache` is `true`, which it is by
+default. The command line option `--cache false` can be used to turn
+off result caching, thus overriding the configuration parameter. If
+`AllCops: UseCache` is set to `false` in the local `.rubocop.yml`,
+then it's `--cache true` that overrides the setting.
+
+The cache is stored in `.rubocop_cache/` under the current directory.
+There are subdirectories for each user, so as to minimize the risk of
+collisions and file curruption from simultaneous `rubocop`
+invocations. Git users should add `.rubocop_cache` to their
```

Comment:
if there are multiple simultaneous `rubocop` runs, it will probably be run by the same user.

- A cruise control system will run as 1 user, and a different directory per branch. This affects a common temp directory but not a local `.rubocop_cache` directory.
- Local developer will run as 1 user. Guard being the use case that may run the same files in same directories, but that seems unlikely.
- Travis will run on different pristine directories, (so caching won't help)

To best handle a race condition, a new cache file could be written and renamed to the target name. last one in wins is better than a corrupt file. But not sure if this really needs to be handled.